Sources/PaymentsCore/Networking/APIClient.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+public class APIClient {
+    public typealias CorrelationID = String
+
+    public var urlSession: URLSession
+    public var environment: Environment
+
+    public init(urlSession: URLSession = .shared, environment: Environment) {
+        self.urlSession = urlSession
+        self.environment = environment
+    }
+
+    public func fetch<T: APIRequest>(
+        endpoint: T,
```

Comment:
I do see how using generics here allows for strongly typed requests. On the other hand, I wonder if the same thing could be achieved with a `protocol` instead?

Reply:
`APIRequest` here is a generic protocol (it has an `associatedtype` constraint), so it can't be referenced as a non-generic function parameter - I think this works for us for the time being since we will know the response type of all the requests we make from our SDK (since the only ones using this `APIClient` would be the other modules in the SDK)

re more loosely typed response - we can definitely add an additional `fetch` function that vends a JSON object instead of a typed result. That would be easy enough to include, but didn't see an immediate reason to add it just yet, so left it off. figured we could add that when we would want to use it instead of now though

Reply:
Got it yeah. We may be able to emulate this Kotlin but it may not be an exact one-to-one correlation. The two languages are similar, but they do diverge at times. We also don't have the standard lib `Result` type that Swift has.

I'm wondering if it would be beneficial to adopt a style that scales across both platforms or if it's best to lean into the strengths of each language when we see fit?

Reply:
I'm doing some due diligence research here. This is a good breakdown of the [Result](https://www.swiftbysundell.com/articles/the-power-of-result-types-in-swift/#:~:text=Using%20a%20Result%20type%20can,while%20retaining%20full%20type%20safety.) type in Swift for the uninitiated. It appears there is a similar [Result](https://medium.com/@jcamilorada/arrow-try-is-dead-long-live-kotlin-result-5b086892a71e) type in Kotlin!

Reply:
Nice! My opinion is that this is an area that we don't strictly tight alignment between platforms, in case this exact sort of API client doesn't make as much sense on Android. If looser typing is a better fit on Android, then I see no problem with deviation.